src/javascript/app_2/App/Containers/Layout/app-contents.jsx

```diff
@@ -4,28 +4,33 @@ import React          from 'react';
 import { withRouter } from 'react-router';
 import { Scrollbars } from 'tt-react-custom-scrollbars';
 import { connect }    from 'Stores/connect';
+import Loading        from '../../../../../templates/app_2/components/loading.jsx';
 
 const AppContents = ({
     children, is_contract_mode,
     is_positions_drawer_on,
     is_blurred,
+    is_loading,
 }) => (
-    <div
-        id='app_contents'
-        className={classNames('app-contents', {
-            'app-contents--show-positions-drawer': is_positions_drawer_on,
-            'app-contents--contract-mode'        : is_contract_mode,
-            'app-contents--is-blurred'           : is_blurred,
-        })}
-    >
-        {/* Calculate height of user screen and offset height of header and footer */}
-        <Scrollbars
-            autoHide
-            style={{ height: 'calc(100vh - 83px)' }}
+    <React.Fragment>
+        { is_loading && <Loading /> }
```

Comment:
Not possible to move the loader logic to app.jsx? and wrap AppContents, Footer, Header

Reply:
as i have tried, it wouldn't work, because it needs to get the store ready that we can use the is_loading